homeassistant/components/opower/config_flow.py

```diff
@@ -88,9 +80,24 @@ async def async_step_user(
             errors = await _validate_login(self.hass, user_input)
             if not errors:
                 return self._async_create_opower_entry(user_input)
-
+        else:
+            user_input = {}
         return self.async_show_form(
-            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_UTILITY,
+                        default=user_input.get(CONF_UTILITY, vol.UNDEFINED),
+                    ): vol.In(get_supported_utility_names()),
+                    vol.Required(
+                        CONF_USERNAME,
+                        default=user_input.get(CONF_USERNAME, vol.UNDEFINED),
+                    ): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
```

Comment:
We could use `self.add_suggested_values_to_schema` instead. For example:

```py
  return self.async_show_form(
      step_id="user",
      data_schema=self.add_suggested_values_to_schema(
          STEP_USER_DATA_SCHEMA,
          user_input
      ),
      errors=errors,
  )
```

It means we can keep the original schema definition that was removed in this pull request as well.

../Frenck

Reply:
Done